### ISSUE TITLE
[Bug]: Page with QR-Code not shown 

### ISSUE THREAD

**Issue by btckai** [btcpayserver/btcpayserver#4959]
### What is your BTCPay version?

1.9.2

### How did you deploy BTCPay Server?

Docker, Umbrel on Linux Ubuntu

### What happened?

I updated BTCPay through Umbrel App-Store. Since then, I can not make a payment through my website. I only use the Key-Pad-Style. On Version 1.7.4 everything worked fine and smooth. But with the current update, I can't settle any payment. I click on the numbers, then on charge, and I see a blank site. On Local Network everything works fine.

LND Version LND 0.16.2-beta 
Umbrel 0.5.3

### How did you encounter this bug?

On Keypad Style I click on the numbers, then on charge, and I see a blank site. On Local Network everything works fine.

### Relevant log output

```shell
sudo docker logs CONTAINER--tail 100
  at BTCPayServer.Hosting.HeadersOverrideMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/HeadersOverrideMiddleware.cs:line 30
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 272
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
info: PayServer:      BTC (Lightning): No more invoice to listen on https://10.21.21.9:8080/, releasing the connection.
fail: Microsoft.AspNetCore.Server.Kestrel: Connection id "0HMQDJFO7KA3O", Request id "0HMQDJFO7KA3O:00000002": An unhandled exception was thrown by the application.
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.Controllers.UIInvoiceController.CreateInvoiceCore(BitpayCreateInvoiceRequest invoice, StoreData store, String serverUrl, List`1 additionalTags, CancellationToken cancellationToken, Action`1 entityManipulator) in /source/BTCPayServer/Controllers/UIInvoiceController.cs:line 113
   at BTCPayServer.Plugins.PointOfSale.Controllers.UIPointOfSaleController.ViewPointOfSale(String appId, Nullable`1 viewType, Nullable`1 amount, String email, String orderId, String notificationUrl, String redirectUrl, String choiceKey, String posData, String formResponse, RequiresRefundEmail requiresRefundEmail, CancellationToken cancellationToken) in /source/BTCPayServer/Plugins/PointOfSale/Controllers/UIPointOfSaleController.cs:line 282
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ExceptionContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.BTCPayMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/BTCpayMiddleware.cs:line 101
   at BTCPayServer.Hosting.GreenfieldMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/GreenfieldMiddleware.cs:line 50
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.HeadersOverrideMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/HeadersOverrideMiddleware.cs:line 30
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
info: PayServer:      BTC (Lightning): Start listening https://10.21.21.9:8080/
fail: Events:         Error while calling event async handler
System.ArgumentException: serviceUrl must be an onion service OR https based OR on the local network (Parameter 'serviceUrl')
   at LNURL.LNURL.EncodeUri(Uri serviceUrl, String tag, Boolean bech32)
   at BTCPayServer.Payments.LNURLPayPaymentType.GetPaymentLink(BTCPayNetworkBase network, InvoiceEntity invoice, IPaymentMethodDetails paymentMethodDetails, Money cryptoInfoDue, String serverUri) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 40
   at BTCPayServer.Payments.LNURLPayPaymentType.PopulateCryptoInfo(InvoiceEntity invoice, PaymentMethod details, InvoiceCryptoInfo invoiceCryptoInfo, String serverUrl) in /source/BTCPayServer/Payments/LNURLPay/PaymentTypes.LNURL.cs:line 62
   at BTCPayServer.Services.Invoices.InvoiceEntity.EntityToDTO() in /source/BTCPayServer/Services/Invoices/InvoiceEntity.cs:line 547
   at BTCPayServer.HostedServices.BitpayIPNSender.Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, Boolean extendedNotification, Boolean sendMail) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 65
   at BTCPayServer.HostedServices.BitpayIPNSender.<StartAsync>b__19_0(InvoiceEvent e) in /source/BTCPayServer/HostedServices/BitpayIPNSender.cs:line 285
   at BTCPayServer.EventAggregator.ChannelSubscription`1.Listen() in /source/BTCPayServer/EventAggregator.cs:line 169
```

### What browser do you use?

Firefox, latest
Chrome latest

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by btckai** [comment]
Updated error log

**Comment by btckai** [comment]
found the problem

**Comment by dennisreimann** [comment]
> found the problem

Can you let us know? :)

**Comment by btckai** [comment]
> > found the problem
> 
> Can you let us know? :)

Problem was, that I just use an HTTP-Address, without SSL. It is just for demonstration. After the Update on Umbrel from BTCPay-Server from 1.7.4. to 1.9 some options changed and got enabled automatically. In this case, LNURL in Lightning Settings got enabled without my doing. This was the source of my error I got displayed. When I disabled LNURL everything works fine.
